### DISCUSSION TITLE
i cannot connect to cloud patch.

### DISCUSSION THREAD

**Discussion by obalola-01** [gitkraken/vscode-gitlens#3151] [Feedback]
i dont know how to edit the settings.json file

**Comment by obalola-01** [comment]
/ Editor settings
    "editor.tabSize": 5,
    "editor.fontSize": 14,
    "editor.insertSpaces": true,
    "editor.wordWrap": "on",
    "editor.fontFamily": "Consolas, 'Courier New', monospace",

    // Theme settings
    "workbench.colorTheme": "Default Dark+",
    "workbench.colorCustomizations": {
        "editorLineNumber.foreground": "#2e4f6c"
    },

    // Language specific settings
    "[typescript]": {
        "editor.formatOnSave": true
    },
    "[python]": {
        "editor.formatOnSave": true
    },

    // Extensions settings
    {
        "extensions.ignoreRecommendations": true,
        "extensions.autoUpdate": "onlySelectedExtensions",
        "editor.bracketPairColorization.enabled": true,
        "editor.codeActionsOnSave": {
            "source.fixAll.eslint": "explicit"{
                "editor.codeActionsOnSave": {
                    "source.fixAll.eslint": true
                    "source.fixAll.eslint": "explicit"
                },
                "eslint.packageManager": "yarn",
                "files.associations": {
                    ".eslintrc*.json": "jsonc"
                },

        },
        }
    }

**Comment by jgonzalez-89** [comment]
It looks like you want to modify your settings.json file for an editor, possibly Visual Studio Code, with specific configurations related to editor behavior, theme, language-specific settings, and extensions. However, the JSON snippet you've provided has some issues that need to be corrected to ensure it's valid JSON. JSON requires proper structure and syntax, including commas between items in objects and arrays, and avoiding dangling commas at the end of lists.

Here is a revised version of your settings.json file with corrections and organization for clarity. Note that I've made assumptions to correct the structure where there were apparent errors or ambiguities:

```json
{
    // Editor settings
    "editor.tabSize": 5,
    "editor.fontSize": 14,
    "editor.insertSpaces": true,
    "editor.wordWrap": "on",
    "editor.fontFamily": "Consolas, 'Courier New', monospace",

    // Theme settings
    "workbench.colorTheme": "Default Dark+",
    "workbench.colorCustomizations": {
        "editorLineNumber.foreground": "#2e4f6c"
    },

    // Language specific settings
    "[typescript]": {
        "editor.formatOnSave": true
    },
    "[python]": {
        "editor.formatOnSave": true
    },

    // Extensions settings
    "extensions.ignoreRecommendations": true,
    "extensions.autoUpdate": "onlySelectedExtensions",
    "editor.bracketPairColorization.enabled": true,
    "editor.codeActionsOnSave": {
        "source.fixAll.eslint": true
    },
    "eslint.packageManager": "yarn",
    "files.associations": {
        ".eslintrc*.json": "jsonc"
    }
}
```

**Reply by obalola-01** [reply]
Thank you so much. i have been trying to correct the code for sometime now. I need a  tutorial on how to edit the settings.json file.

**Reply by jgonzalez-89** [reply]
How to Edit settings.json in Visual Studio Code:

- Open Command Palette: Press Ctrl+Shift+P (Windows/Linux) or Cmd+Shift+P (macOS) to open the Command Palette.
- Open Settings (JSON): Type Open Settings (JSON) and select the command that appears. This will open your settings.json file.
- Edit and Save: Paste the corrected JSON into your settings.json file, replacing the existing content or merging it as needed. Save the file with Ctrl+S (Windows/Linux) or Cmd+S (macOS).

**Reply by obalola-01** [reply]
that is the problem. I dont know how to edit the file. There are bunch of extensions that needs to be in the settings.json file.

**Reply by obalola-01** [reply]
can you be my mentor?

**Reply by jgonzalez-89** [reply]
Open visual studio code, and press Ctrl+Shift+P (Windows/Linux) or Cmd+Shift+P (macOS), type Type Open Settings and select Type Open Settings (JSON), 

<img width="1079" alt="Captura de pantalla 2024-02-16 a las 22 55 21" src="https://github.com/gitkraken/vscode-gitlens/assets/69685060/4777d5b1-b6d6-4617-a914-3da37fa4c0e0">


you will see a settings.json file open, edit it to have these settings

```python
{
    // Editor settings
    "editor.tabSize": 4,
    "editor.fontSize": 10,
    "editor.insertSpaces": true,
    "editor.wordWrap": "on",
    "editor.fontFamily": "Consolas, 'Courier New', monospace",

    // Theme settings
    "workbench.colorTheme": "Default Dark+",
    "workbench.colorCustomizations": {
        "editorLineNumber.foreground": "#2e4f6c"
    },

    // Language specific settings
    "[typescript]": {
        "editor.formatOnSave": true
    },
    "[python]": {
        "editor.formatOnSave": true
    },

    // Extensions settings
    "extensions.ignoreRecommendations": true,
    "extensions.autoUpdate": "onlySelectedExtensions",
    "editor.bracketPairColorization.enabled": true,
    "editor.codeActionsOnSave": {
        "source.fixAll.eslint": true
    },
    "files.associations": {
        ".eslintrc*.json": "jsonc"
    }
}
```

<img width="671" alt="Captura de pantalla 2024-02-16 a las 22 58 25" src="https://github.com/gitkraken/vscode-gitlens/assets/69685060/34644753-92c6-4713-b1a7-4b7b5b9c7a5b">


save

<img width="659" alt="Captura de pantalla 2024-02-16 a las 23 04 53" src="https://github.com/gitkraken/vscode-gitlens/assets/69685060/0fc173e4-b849-45fb-8b60-d773ff2e098b">

**Reply by obalola-01** [reply]
Thank you very much. Are you in Canada?

**Comment by obalola-01** [comment]
I just need your guidance. I don't know how to edit code

**Reply by jgonzalez-89** [reply]
Please check this video, it may help you.


https://www.youtube.com/watch?v=NXGIGU1jIXg&ab_channel=ProgrammingKnowledge

**Reply by obalola-01** [reply]
MR Gonzalez please help me check this settings. json file. thank you

**Comment by obalola-01** [comment]
Thanks, I'll check it out.

**Reply by obalola-01** [reply]
/ Editor settings
    "editor.tabSize": 4,
    "editor.fontSize": 10,
    "editor.insertSpaces": true,
    "editor.wordWrap": "on",
    "editor.fontFamily": "Consolas, 'Courier New', monospace",

    // Theme settings
    "workbench.colorTheme": "Default Dark+",
    "workbench.colorCustomizations": {
        "editorLineNumber.foreground": "#2e4f6c"
    },

    // Language specific settings
    "[typescript]": {
        "editor.formatOnSave": true
    },
    "[python]": {
        "editor.formatOnSave": true
    },

    // Extensions settings
    "extensions.ignoreRecommendations": true,
    "vsintellicode.python.completionsEnabled": true,
    "vsintellicode.typescript.completionsEnabled": true,
    "editor.bracketPairColorization.enabled": true,
    "explorer.excludeGitIgnore": false,
    "editor.codeActionsOnSave": {
     "source.organizeImports": true,
     "source.fixAll": true,
     "source.fixAll.eslint": true
    },
    "files.associations": {
        ".eslintrc*.json": "jsonc"
    }
}

the settings.json seems to be correct. can you help me check for error.  thank you

**Comment by jkelroy** [comment]
@obalola-01 taking a step back, can you expand on the issue you are having with Cloud Patches? Are you simply unable to see the Cloud Patch section in the Side Bar or are you having an issue working with them?

If you are unable to see them, can you make sure you are on the latest version of GitLens (14.8.2)? You can check if there are updates available from Extensions.

<img width="392" alt="Screenshot 2024-02-22 at 4 07 16 PM" src="https://github.com/gitkraken/vscode-gitlens/assets/27739784/3e658b50-08e1-47e9-ab64-cf829cc3ae0d">

Notably, [Cloud Patches](https://help.gitkraken.com/gitlens/gitlens-features/#cloud-patches-preview-%e2%98%81%ef%b8%8f) are enabled by default so you do not have to adjust the settings. They can be seen from the GitLens sidebar and it can be focused from the command palette - command/ctrl + shift + P > _Gitlens: Focus on Cloud Patches View_.

![Screenshot 2024-02-22 at 4 11 39 PM](https://github.com/gitkraken/vscode-gitlens/assets/27739784/f52299dd-12be-4d68-b0fc-31c7a727fd06)

If you do not see Cloud Patches, there is a chance that you have manually disabled them by setting `gitlens.cloudPatches.endabled` to `false`. To check this and/or set it to true:
1. Open the command palette with command/ctrl + shift + P > _Preferences: open user settings (JSON)_
2. Set `{"gitlens.cloudPatches.endabled": true}`
3. Save the file

![Screenshot 2024-02-22 at 4 16 27 PM](https://github.com/gitkraken/vscode-gitlens/assets/27739784/71c348b8-31bf-4c21-84d3-03c9a23013d6)

Lastly, you do need to be logged into GitLens with a a trial or paid account to use with Cloud Patches.

**Comment by obalola-01** [comment]
Cloud patches is working fine. I was having issue with the settings.json
file. But I am still on it. Thank you
